source/uwp/Renderer/lib/MediaHelpers.cpp

```diff
@@ -181,22 +181,49 @@ namespace AdaptiveCards::Rendering::Uwp::MediaHelpers
 
     void HandleMediaResourceResolverCompleted(winrt::IAsyncOperation<winrt::IRandomAccessStream> const& operation,
                                               winrt::AsyncStatus status,
-                                              winrt::MediaElement const& mediaElement,
-                                              winrt::hstring const& mimeType)
+                                              winrt::MediaPlayerElement const& mediaElement,
+                                              winrt::hstring const& mimeType,
+                                              winrt::AdaptiveMedia const& adaptiveMedia,
+                                              winrt::AdaptiveRenderContext const& renderContext)
     {
         if (status == winrt::AsyncStatus::Completed)
         {
             // Get the random access stream
             if (const auto randomAccessStream = operation.GetResults())
             {
-                mediaElement.SetSource(randomAccessStream, mimeType);
+                auto mediaSrc = winrt::MediaSource::CreateFromStream(randomAccessStream, mimeType);
+                if (adaptiveMedia.CaptionSources().Size() > 0)
+                {
+                    for (const auto captionSource : adaptiveMedia.CaptionSources())
+                    {
+                        const auto timedTextURL = GetUrlFromString(renderContext.HostConfig(), captionSource.Url());
+                        const auto timedTextSrc = winrt::TimedTextSource::CreateFromUri(timedTextURL);
+                        timedTextSrc.Resolved(
+                            [captionSource](winrt::TimedTextSource sender, winrt::TimedTextSourceResolveResultEventArgs args)
+                            {
+                                if (!args.Error())
+                                {
+                                    args.Tracks().GetAt(0).Label(captionSource.MimeType());
```

Comment:
`Label` corresponds to user friendly name of the track - in this case the language of this subtitles track, not the **format** of the file :)
We also need to validate the `mimeType` (extension) to match one of the ones we allow like we do it for `MediaSource`: [supported mimeTypes](https://github.com/microsoft/AdaptiveCards/blob/3a35997792bd7f7b4d4deb155f21804803084e70/source/uwp/Renderer/lib/MediaHelpers.cpp#L10) and [how it's being used](https://github.com/microsoft/AdaptiveCards/blob/main/source/uwp/Renderer/lib/MediaHelpers.cpp#L150-L180)

Reply:
`Label` is now set to the language property, and we validate the `mimeType` now